p4rt_client/p4rt_client.go

```diff
@@ -46,6 +50,27 @@ const (
 	P4RT_STREAM_TERM_CHAN_SIZE      = 100
 )
 
+// flagCred implements credentials.PerRPCCredentials by populating the
+// username and password metadata from flags.
+type flagCred struct {
+	username string
+	password string
+}
+
+// GetRequestMetadata is needed by credentials.PerRPCCredentials.
+func (s flagCred) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
+	// We can pass any metadata to the server here
+	return map[string]string{
+		"username": s.username,
+		"password": s.password,
+	}, nil
+}
+
+// RequireTransportSecurity is needed by credentials.PerRPCCredentials.
+func (s flagCred) RequireTransportSecurity() bool {
+	return false
```

Comment:
Shouldn't this return true if using TLS? maybe add a 'secure' field to flagCred?

Reply:
I am not sure if this matters, because the calling function would have set TLS if needed (and we should not require it). But in any case, i have added a flag.